cocotb/scheduler.py

```diff
@@ -231,9 +232,10 @@ def __init__(self):
         # Our main state
         self._mode = Scheduler._MODE_NORMAL
 
-        # A dictionary of pending (write_func, args), keyed by handle. Only the last scheduled write
-        # in a timestep is performed, all the rest are discarded in python.
-        self._write_calls = _py_compat.insertion_ordered_dict()
+        # A dictionary of pending (write_func, args), keyed by handle.
+        # Writes are applied oldest to newest (least recently used).
+        # Only the last scheduled write to a particular handle in a timestep is performed.
+        self._write_calls = OrderedDict()
```

Comment:
We only support cpython 3.6+, which guarantees insertion ordered dicts (the Python "language" guarantees it since 3.7). You can simply use `{}` here.

Reply:
Normal dicts don't support `.popitem(last=False)`.

Reply:
oh indeed.